src/tables.jl

```diff
@@ -66,13 +47,16 @@ Base.getproperty(t::Table, s::Symbol) = [getproperty(iterate(t, i)[1], s) for i
 function Base.getindex(t::Table, idx::Int)
     layer = getlayer(t)
     setnextbyindex!(layer, idx) 
+    return get_row(layer)
+end
+    
+function get_row(layer::IFeatureLayer)
     featuredefn = layerdefn(layer)
-    field_names = Tuple(Symbol(getname(getfielddefn(featuredefn, i-1))) for i in 1:nfield(layer))
     fielddefns = (getfielddefn(featuredefn, i) for i in 0:nfield(layer)-1)
-    types_fields = (_FIELDTYPE[gettype(fielddefn)] for fielddefn in fielddefns)
-    geom_names = Tuple(Symbol(getname(getgeomdefn(featuredefn, i-1))) for i in 1:ngeom(layer))
-    v = Union{types_fields..., IGeometry}[]
+    field_names = Tuple(Symbol(getname(fielddefn)) for fielddefn in fielddefns)
+    geom_names = (Symbol(getname(getgeomdefn(featuredefn, i-1))) for i in 1:ngeom(layer))
 
+    v = Union{Tables.schema(layer).types...}[]
```

Comment:
It's work looking if it's better to just make it `Any`, i.e. `v = []`. Or alternatively, is it possible to do this without creating a vector, for instance making it a generator and converting that to Tuple.